homework/index.js

```diff
@@ -36,11 +36,115 @@
+        const basicInfo = document.getElementById('basicInfo');
+
+        // the header of the the basic information repository.
+        createAndAppend('h2', basicInfo, { text: 'Repository Info' });
+
+        // button to hold the name of the selected repository.
+        createAndAppend('button', basicInfo, {
+          id: 'repo-name-btn',
+          value: '',
+        });
+        const repoNameBtn = document.getElementById('repo-name-btn');
+
+        // list of repository basic information.
+        createAndAppend('ul', basicInfo, { id: 'info-ul' });
+        const infoList = document.getElementById('info-ul');
+
+        // select element to select the wanted repository.
+        createAndAppend('select', container, { id: 'select' });
+        const select = document.getElementById('select');
+
+        // the repository contributors (right aside).
+        createAndAppend('aside', container, {
+          class: 'right-aside box',
+          id: 'contributorsAside',
+        });
+        const contributorsAside = document.getElementById('contributorsAside');
+
+        // the header of the contributors aside.
+        createAndAppend('h2', contributorsAside, { text: 'Contributor Info' });
+
+        // ordered list of contributors of each repository selected.
+        createAndAppend('ol', contributorsAside, { id: 'contributors-ol' });
+        const contributorsList = document.getElementById('contributors-ol');
+
+        // sorting the repositories (case-insensitive) on repository name.
+        data.sort((a, b) => a.name.localeCompare(b.name, 'en', { sensitivity: 'base' }));
+        data.forEach(repo => {
+          createAndAppend('option', select, { text: repo.name });
+        });
+
+        // show the first repository basic info on the left aside & the contributors on the right one.
+        const selected = document.getElementById('select');
+
+        const showInfo = function() {
+          data.forEach(repo => {
+            if (selected.value === repo.name) {
+              // basic information (left aside)
+              repoNameBtn.innerText = '';
+              createAndAppend('a', repoNameBtn, {
+                text: repo.name,
+                href: repo.html_url,
+                target: '_blank',
+                id: repo.id + 1,
+              });
+              createAndAppend('li', infoList, { text: repo.id, class: 'list' });
+              createAndAppend('li', infoList, {
+                text: repo.language,
+                class: 'list',
+              });
+              // contributors (right aside)
+              fetchJSON(repo.contributors_url, (error, contributorsData) => {
+                if (error) {
+                  createAndAppend('div', root, {
+                    text: error.message,
+                    class: 'alert-error',
+                  });
+                } else {
+                  contributorsList.innerText = '';
+                  contributorsData.forEach(contributor => {
```

Comment:
There is a small JS error here. For some repos, if they aren't having any contributors list i.e `contributorsData` is `null`, `forEach` would throw an error as it can't loop over a null element. 🙂